psydac/fem/basic.py

```diff
@@ -125,29 +141,54 @@ def eval_field_gradient( self, field, *eta , weights=None):
     # Concrete methods
     #----------------------
     def __mul__(self, a):
-        from psydac.fem.vector import ProductFemSpace
+        from psydac.fem.vector import create_product_space
 
         spaces = [*(self.spaces if self.is_product else [self]),
                   *(   a.spaces if    a.is_product else    [a])]
  
-        space = ProductFemSpace(*spaces)
+        space = create_product_space(*spaces)
         if a.symbolic_space and self.symbolic_space:
             space._symbolic_space =  self.symbolic_space*a.symbolic_space
         return space
 
     # ...
     def __rmul__(self, a):
-        from psydac.fem.vector import ProductFemSpace
+        from psydac.fem.vector import create_product_space
 
         spaces = [*(   a.spaces if    a.is_product else    [a]),
                   *(self.spaces if self.is_product else [self]),]
 
-        space = ProductFemSpace(*spaces)
+        space = create_product_space(*spaces)
 
         if a.symbolic_space and self.symbolic_space:
             space._symbolic_space =  a.symbolic_space * self.symbolic_space
         return space
```

Comment:
I am quite puzzled by these two concrete methods, which I cannot recognize. Apparently this code was added by @saidctb 4 years ago when adding support for repeated knots.

In principle, since `FemSpace` is an abstract base class, it should not have any concrete methods. In fact, these two methods are inconsistent because there is no guarantee that a `FemSpace` object will have a `spaces` attribute.

Accordingly, the new properties `patch_spaces` and `component_spaces` cannot be defined in this class as concrete. They currently use the attribute `_spaces` which does not exist.

Reply:
The tests seem to pass. I suggest to leave these functions raising a NotImplementedError and if we discover that some code is using them, we'll look for a good solution.

Reply:
Good idea!